### PR TITLE
Make optimal pipeline in Exercise M3.02 depend on preprocessing strategy

### LINKED ISSUE #225: Add figure of the effect of StandardScaler in the M2 wrap quiz correction
@ogrisel proposes to add an additional figure: https://gitlab.inria.fr/learninglab/mooc-scikit-learn/mooc-scikit-learn-coordination/-/merge_requests/8#note_477599 

It is probably not dramatic for the next course but we should address it.
I am putting this issue as a reminder.

### LINKED ISSUE #646: Optimal pipeline in Exercise M3.02 does not depend on preprocessing strategy
As discussed in [this forum post](https://mooc-forums.inria.fr/moocsl/t/exercise-m3-02-scaling/8179), the narrative in the [Exercise M3.02 notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_sol_03.html) does not correspond to the best model found by a `RandomizedSearchCV` nor the results shown with more iterations in the parallel coordinate plot. The reason seems to be that the best performing models with different preprocessing strategies overlap considerably.

One possible solution is to keep the narrative of the exercise but use another dataset that has variables with more diverse ranges of values. In that case, we can use the opportunity to add a dataset to the MOOC as suggested in [#271](https://github.com/INRIA/scikit-learn-mooc/issues/271) and [this forum post](https://mooc-forums.inria.fr/moocsl/t/grateful-but-conflicted/10934).

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
Fixes #646.
Fixes #225.

This PR reworks [Exercise M3.02 notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_sol_03.html) to use `california_housing` instead of `adult_census` and a `KNeighborsRegressor` instead of `LogisticRegression`. In this way the pipeline is more sensitive to scaling without the need of a dataset external to the present contents of the MOOC.

A side effect is that the example gets simplified in terms of both complexity and computation time, and then there is no longer need to pre-compute a csv file with the `cv_results`, nor to limit the size of the training set.

A possible downside of this PR is that introducing too many details of how KNeighbors models work may be distracting. Maybe we can remove the plots with the distances as annotations.

### CODE REVIEW

**Review by ogrisel — COMMENTED** [review]
I like the switch to the new dataset and model. Here are further suggestions. Some of them were made on the exercise file but I meant to do them on the solution file to resync the exercise from it afterwards.

File: python_scripts/parameter_tuning_ex_03.py [hunk 1]
```diff
@@ -17,85 +17,51 @@
...
-adult_census = pd.read_csv("../datasets/adult-census.csv")
-
-target_name = "class"
-target = adult_census[target_name]
-data = adult_census.drop(columns=[target_name, "education-num"])
+from sklearn.datasets import fetch_california_housing
 from sklearn.model_selection import train_test_split
 
+data, target = fetch_california_housing(return_X_y=True, as_frame=True)
+target *= 100  # rescale the target in k$
```
[the remark this answers is not in the document]
Comment (ArturoAmorQ) [reply python_scripts/parameter_tuning_ex_03.py:26]: It's rather for consistency with the rest of the notebooks where we use California housing.

File: python_scripts/parameter_tuning_sol_03.py [hunk 2]
```diff
@@ -183,14 +145,93 @@
...
+        scaled_data.iloc[[0, i]]["Population"],
+        "k--",
+        linewidth=1,
+    )
+    plt.annotate(
+        scaled_dist[0, i],
+        (scaled_data.iloc[i]["AveRooms"], scaled_data.iloc[i]["Population"]),
+    )
+plt.axis("square")
+_ = plt.title("Distances to sample_of_interest after scaling")
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply python_scripts/parameter_tuning_sol_03.py:221]: @ArturoAmorQ I assume that you want to remove the plot?

[the remark this answers is not in the document]
Comment (ArturoAmorQ) [reply python_scripts/parameter_tuning_sol_03.py]: Yes, I removed the plot as suggested by @ogrisel.

**Review by glemaitre — APPROVED** [review]
Feel free to merge when you want.

**Pull request merged**